Hivemind/Assets/SpatialPartition.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SpatialPartition : MonoBehaviour
+{
+
+    [SerializeField]
+    private GameObject obj;
+
+    [SerializeField]
+    private int width;
+    [SerializeField]
+    private int depth;
+    // Start is called before the first frame update
+    void Start()
+    {
+        for (int i = 0; i < width; i++)
+        {
+            for (int j = 0; j < depth; j++)
+            {
+                GameObject go = Instantiate(obj, transform, false);
+                go.transform.position = new Vector3(
+                    (0.0f - transform.localScale.x / 2.0f) + ((transform.localScale.x / width) / 2.0f) + (transform.localScale.x / width) * i,
```

Comment:
Those brackets are there for readability reasons, I feel like it would be best to leave those in to make it easier to work with. They separate calculations from each other so you know what is calculated "localy"